packages/augur-core/source/contracts/trading/ZeroXTrade.sol

```diff
@@ -351,6 +351,16 @@ contract ZeroXTrade is Initializable, IZeroXTrade, IERC1155 {
             address _kycToken
         )
     {
+         // Read the bytes4 from array memory
+        assembly {
+            _assetProxyId := mload(add(_assetData, 32))
+            // Solidity does not require us to clean the trailing bytes.
+            // We do it anyway
```

Comment:
Why do you not have to do the logical and here?  Seems necessary to me...

Reply:
My understanding is the bytes4 value will not overflow but will just cut off the extra

Reply:
Since you do the check anyway, I'll accept that statement at face value.  🙂

Reply:
That seems magical you don't have to do it. Assembly doesn't know about the data type, so does non-assembly always mask the data it pulls off the stack?

Reply:
Seems so.

```
contract Test {

    function decodeAssetData(bytes memory _assetData)
        public
        pure
        returns (
            bytes4 _assetProxyId
        )
    {
        assembly {
            _assetProxyId := mload(add(_assetData, 32))
            _assetProxyId := and(_assetProxyId, 0xFFFFFFFF00000000000000000000000000000000000000000000000000000000)
        }
        
        require(_assetProxyId == "0x22222222");

        return _assetProxyId;
    }
}
```

for example when passed "0x22222222" will succeed and return the expected result.

Reply:
```
contract Test {
    function decodeAssetData(bytes memory _assetData) public pure returns (bytes1 _assetProxyId)
    {
        assembly {
            _assetProxyId := mload(add(_assetData, 32))
        }
        require(_assetProxyId == 0xff);
        return _assetProxyId;
    }
}
```
I think this is the code that is surprising to @epheph and I.  If you pass in `0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` I didn't expect the `require` check to pass, though now that I see it in action I suspect I know _why_ it passes (assignment to `bytes1` effectively does the truncation for you).

Reply:
oh I left the masking in heh. Yea it does the magic.